langchain/llms/base.py

```diff
@@ -184,15 +195,32 @@ async def agenerate(
             missing_prompts,
         ) = get_prompts(params, prompts)
         if len(missing_prompts) > 0:
-            self.callback_manager.on_llm_start(
-                {"name": self.__class__.__name__}, missing_prompts, verbose=self.verbose
-            )
+            if self.callback_manager.is_async:
+                await self.callback_manager.on_llm_start(
+                    {"name": self.__class__.__name__},
+                    missing_prompts,
+                    verbose=self.verbose,
+                )
+            else:
+                self.callback_manager.on_llm_start(
```

Comment:
FYI @nfcampos @hwchase17 I decided to call the callback manager synchronously instead of running it in the threadpool executor because thread safety is a concern. For example, the default callback manager already includes a callback handler `StdOutCallbackHandler` that prints to stdout, having multiple threads trying to print to stdout at the same time is not great.

Reply:
See my other comment. Not sure yet about this one, but I'd be tempted to forbid using async chains with sync callbackmanagers for simplicity, thoughts?